typescript/deploy/src/deploy.ts

```diff
@@ -48,17 +50,33 @@ export abstract class AbacusDeployer<
 
   abstract deployContracts(chain: Chain, config: Config): Promise<Contracts>;
 
-  async deploy() {
-    this.logger('Start Deploy');
-    this.verificationInputs = objMap(this.configMap, () => []);
-    const chains = this.multiProvider.chains();
-    const entries: [Chain, Contracts][] = [];
-    for (const chain of chains) {
+  async deploy(
+    partialDeployment: Partial<Record<Chain, Contracts>> = this
+      .deployedContracts,
+  ): Promise<Record<Chain, Contracts>> {
+    const deployedChains = Object.keys(partialDeployment);
+    const configChains = Object.keys(this.configMap);
+    const targetChains = this.multiProvider
```

Comment:
This might be more related to #394, but imo there should be some indication when the multiProviders chains are not covering the configChains, happy to be out of scope here